src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -162,27 +149,101 @@ unsigned int MethodContext::saveToFile(HANDLE hFile)
 #define LWM(map, key, value) sparseWriteFileCR(map);
 #include "crlwmlist.h"
 
-    // Write the end canary
+    // Add canary
     buff2[buffIndex++] = '4';
     buff2[buffIndex++] = '2';
+    Assert(buffIndex == totalLen);
 
-    Assert(buffIndex == totalFileSize);
+    return data;
```

Comment:
Does this do a full copy of the array? If so, that could be expensive.

Reply:
No, even if RVO is not used I believe the copy should be elided and a move operation will happen instead.